lib/assignable_values/active_record/restriction/scalar_attribute.rb

```diff
@@ -10,17 +10,21 @@ def initialize(*args)
           define_humanized_assignable_values_instance_method
         end
 
-        def humanized_value(value)
+        def humanized_value(klass, value)
           if value.present?
-            dictionary_scope = "assignable_values.#{model.name.underscore}.#{property}"
-            I18n.t(value, :scope => dictionary_scope, :default => default_humanization_for_value(value))
+            klass.lookup_ancestors.select(&:name).each do |klass|
+              dictionary_scope = :"assignable_values.#{klass.model_name.i18n_key}.#{property}"
+              translation = I18n.translate(value, scope: dictionary_scope, default: nil)
+              return translation unless translation.nil?
+            end
+            default_humanization_for_value(value)
```

Comment:
I prefer to not `return` from a block. It returns from the method instead of the block, which is what we need here, but it's also an unintuive Ruby quirk that I try to avoid where possible.

What do you think about this instead?

```ruby
def humanized_value(klass, value)
  if value.present?
    humanization_from_i18n(klass, value) || default_humanization_for_value(value)
  end
def 

def humanization_from_i18n(klass, value)
  klass.lookup_ancestors.select(&:name).detect do |klass|
    dictionary_scope = :"assignable_values.#{klass.model_name.i18n_key}.#{property}"
    I18n.translate(value, scope: dictionary_scope, default: nil)
  end
end
```

Reply:
Yes, I know that this is not very pretty. `detect` however will give me the Class that has a translation, not the translation itself. I don't want to translate the same value twice.
I will use the approch described in [this card](https://makandracards.com/makandra/35569-how-to-iterate-over-an-enumerable-returning-the-first-truthy-result-of-a-block-map-find), i.e. using `break` inside the block to detect the translation.